clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -95,6 +103,31 @@ static Attr *handleIntelFPGALoopAttr(Sema &S, const ParsedAttr &A) {
     }
   }
 
+  if (A.getParsedKind() == ParsedAttr::AT_SYCLIntelFPGAII &&
+      checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion) << "'intel::ii'";
+  } else if (A.getKind() == ParsedAttr::AT_SYCLIntelFPGAMaxConcurrency &&
+             checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::max_concurrency'";
+  } else if (A.getKind() == ParsedAttr::AT_SYCLIntelFPGAMaxConcurrency &&
+             checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::max_concurrency'";
+  } else if (A.getKind() == ParsedAttr::AT_SYCLIntelFPGAMaxInterleaving &&
+             checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::max_interleaving'";
+  } else if (A.getKind() == ParsedAttr::AT_SYCLIntelFPGASpeculatedIterations &&
+             checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::speculated_iterations'";
+  } else if (A.getKind() == ParsedAttr::AT_SYCLIntelFPGALoopCoalesce &&
+             checkDeprecatedSYCLLoopAttributeSpelling(S, A)) {
+    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
+        << "'intel::loop_coalesce'";
+  }
```

Comment:
Is else conditions necessary here?

Reply:
> Is else conditions necessary here?

I do not think that we need else here since we give separate warning/error for unknow loop attribute spellings. If you would want, I can split the "else if " to 
else {
 if (checkDeprecatedSYCLLoopAttributeSpelling(S, A))    
    S.Diag(A.getLoc(), diag::note_spelling_suggestion)
        << "'intel::loop_coalesce'";
  }

Reply:
I think I'm fine with the current state.

Reply:
Thanks Mariya.